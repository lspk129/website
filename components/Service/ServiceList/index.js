import React from 'react';
import styled from 'styled-components';

import ServiceContainer from '../ServiceContainer';

const StyledSection = styled.div`
  transition: all 0.5s ease-in-out;
  &:hover {
    background: ${props => props.theme.background};
    color: ${props => props.theme.text};
    .material-icons {
      color: ${props => props.theme.text};
    }
  };
  .material-icons {
    transition: all 0.5s ease-in-out;
    font-size: 48px;
    padding: 20px ;
    color: ${props => props.theme.background};
  }
  p {
    padding: 10px 50px;
  }
`;

const ServiceList = () => (
  <ServiceContainer>
    <StyledSection>
      <div><i className="material-icons">desktop_mac</i></div>
      <h3>Web development</h3>
      <p>We help you build fast, functional, performance oriented websites and web apps</p>
    </StyledSection>
    <StyledSection>
      <div><i className="material-icons">color_lens</i></div>
      <h3>Web design</h3>
      <p>We are creating unique and adaptive designs to engage customers</p>
    </StyledSection>
  </ServiceContainer>
);

export default ServiceList;
