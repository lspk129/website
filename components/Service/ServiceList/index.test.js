import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import ServiceList from './index';

describe('ServiceList', () => {
  it('renders corretcly', () => {
    const wrapper = renderer.create(<ServiceList />);
    const tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders image of service', () => {
    const wrapper = shallow(<ServiceList />);
    expect(wrapper).toBePresent();
    expect(wrapper).not.toHaveTagName('i');
  });

  it('renders h3 element with content "Web development"', () => {
    const wrapper = shallow(<ServiceList />);
    expect(wrapper).toBePresent();
    expect(wrapper).not.toHaveHTML('<h3>Web development</h3>');
  });

  it('renders p element with content "We help you build fast, functional, performance oriented websites and web apps"', () => {
    const wrapper = shallow(<ServiceList />);
    expect(wrapper).toBePresent();
    expect(wrapper).not.toHaveHTML(
      '<p>We help you build fast, functional, performance oriented websites and web apps</p>',
    );
  });
});
