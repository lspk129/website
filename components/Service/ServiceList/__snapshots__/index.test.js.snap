// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceList renders corretcly 1`] = `
<div
  className="sc-bdVaJa jFsOXn"
>
  <div
    className="sc-bwzfXH kxVrmy"
  >
    <div>
      <i
        className="material-icons"
      >
        desktop_mac
      </i>
    </div>
    <h3>
      Web development
    </h3>
    <p>
      We help you build fast, functional, performance oriented websites and web apps
    </p>
  </div>
  <div
    className="sc-bwzfXH kxVrmy"
  >
    <div>
      <i
        className="material-icons"
      >
        color_lens
      </i>
    </div>
    <h3>
      Web design
    </h3>
    <p>
      We are creating unique and adaptive designs to engage customers
    </p>
  </div>
</div>
`;
